src/main/java/appeng/thirdparty/codechicken/lib/model/pipeline/transformers/QuadCornerKicker.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2018 covers1624
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package appeng.thirdparty.codechicken.lib.model.pipeline.transformers;
+
+
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumFacing.AxisDirection;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.Vec3i;
+
+import appeng.thirdparty.codechicken.lib.model.Quad.Vertex;
+import appeng.thirdparty.codechicken.lib.model.pipeline.IPipelineElementFactory;
+import appeng.thirdparty.codechicken.lib.model.pipeline.QuadTransformer;
+
+import static net.minecraft.util.EnumFacing.AxisDirection.NEGATIVE;
+import static net.minecraft.util.EnumFacing.AxisDirection.POSITIVE;
+
+
+/**
+ * This transformer is a little complicated.
+ * Basically a Facade / Cover can use this to 'kick' the edges
+ * in of quads to fix z-Fighting in the corners.
+ * Use it by specifying the side of the block you are on,
+ * the bitmask for where the other Facades / Cover's are,
+ * the bounding box of the facade, NOT the hole piece,
+ * and the thickness of your Facade / Cover, this is used
+ * as the kick amount.
+ *
+ * @author covers1624
+ */
+public class QuadCornerKicker extends QuadTransformer
+{
+
+	//The factory for pipeline creation.
+	public static final IPipelineElementFactory<QuadCornerKicker> FACTORY = QuadCornerKicker::new;
+
+	//Simple horizonal lookups.
+	public static int[][] horizonals = new int[][] {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this "public static horizonals" field final [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1444)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make horizonals a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this member "protected". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2386)